minmax_stack.rb

```diff
@@ -4,37 +4,60 @@
 # "maximum" _and_ "minimum" operations.
 
 
+
 class MinMaxStack
+
+  class DimensionError < StandardError;end
+
+  attr_accessor :items
+
   def initialize
+    @list = LinkedList.new
+    @items = []
   end
 
   # Places +item+ on the top of the stack
   def push(item)
+    @list.unshift(item)
+    @items.push(item)
+
+    self
   end
 
   # Removes the item on the top of the stack and returns it.
   # Raises an error if the stack is empty (called a "stack underflow")
   def pop
+    if self.empty?
+      raise DimensionError, "The stack is empty!"
+    else
+      @items = @items - [@list.head.value]
```

Comment:
Oh.  Well, can't use a `LinkedList` if you're going to do this, obviously! :smile: